src/HttpClient.js

```diff
@@ -49,9 +49,10 @@ class HttpClient {
     headers={},
     attempts=0,
     failover=true,
-    forceFailover=false
+    forceFailover=false,
+    nodeUrl
   }) {
-    let baseURI = this.BaseURI();
+    let baseURI = this.BaseURI(nodeUrl);
```

Comment:
Suggest add comment above this line e.g. `// if nodeUrl passed in, restrict communication to that node only (unless previously recorded write token is found in next step)`